tdigest.go

```diff
@@ -335,3 +345,96 @@ func (cs *centroidSet) MergeInto(other TDigest) {
 		other.Add(c.mean, c.count)
 	}
 }
+
+func (cs *centroidSet) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(nil)
+	w := &stickyErrWriter{w: buf, err: nil}
+	binary.Write(w, binary.LittleEndian, int32(1))
+
+	var min, max float64
+	if len(cs.centroids) > 0 {
+		min = cs.centroids[0].mean
+		max = cs.centroids[len(cs.centroids)-1].mean
+	}
+
+	binary.Write(w, binary.LittleEndian, min)
+	binary.Write(w, binary.LittleEndian, max)
+	binary.Write(w, binary.LittleEndian, float64(cs.compression))
+	binary.Write(w, binary.LittleEndian, int32(len(cs.centroids)))
+	if w.err != nil {
+		return nil, w.err
+	}
+	for _, c := range cs.centroids {
+		binary.Write(w, binary.LittleEndian, int64(c.count))
+		binary.Write(w, binary.LittleEndian, float64(c.mean))
+	}
+	if w.err != nil {
+		return nil, w.err
+	}
+	return buf.Bytes(), nil
+}
+
+func (cs *centroidSet) UnmarshalBinary(p []byte) error {
+	var (
+		encodingVersion int32
+		min, max        float64
+		n               int32
+	)
+	r := &stickyErrReader{r: bytes.NewReader(p), err: nil}
+
+	binary.Read(r, binary.LittleEndian, &encodingVersion)
+	if encodingVersion != 1 {
+		return fmt.Errorf("invalid encoding version: %d", encodingVersion)
+	}
+
+	binary.Read(r, binary.LittleEndian, &min)
+	binary.Read(r, binary.LittleEndian, &max)
```

Comment:
The values of `min` and `max` aren't used here. Do they need to be part of the serialization?

Reply:
I'm just matching the behavior of the java reference implementation's serialization, here.